src/compiler/tsbuildPublic.ts

```diff
@@ -64,12 +64,13 @@ namespace ts {
         return existingValue || newValue!;
     }
 
-    function getOrCreateValueMapFromConfigFileMap<T>(configFileMap: ESMap<ResolvedConfigFilePath, ESMap<string, T>>, resolved: ResolvedConfigFilePath): ESMap<string, T> {
-        return getOrCreateValueFromConfigFileMap<ESMap<string, T>>(configFileMap, resolved, () => new Map());
+    function getOrCreateValueMapFromConfigFileMap<K extends string, V>(configFileMap: ESMap<ResolvedConfigFilePath, ESMap<K, V>>, resolved: ResolvedConfigFilePath): ESMap<K, V> {
+        return getOrCreateValueFromConfigFileMap(configFileMap, resolved, () => new Map());
     }
 
-    function newer(date1: Date, date2: Date): Date {
-        return date2 > date1 ? date2 : date1;
+    /*@internal*/
+    export function getCurrentTime(host: { now?(): Date; }) {

```

Comment:
This is used everywhere to ensure we are using host.now method if present during testing to ensure we arent using real current time in baselines